substrate/frame/support/src/traits/messages.rs

```diff
@@ -170,17 +182,13 @@ pub trait QueueFootprintQuery<Origin> {
 	/// Return the state footprint of the given queue.
 	fn footprint(origin: Origin) -> QueueFootprint;
 
-	/// Check if the number of resulting new pages after enqueuing the provided messages respects
-	/// the provided limit.
-	///
-	/// On success, it returns the number of resulting new pages.
-	/// On error, it returns a tuple containing the number of resulting new pages and the index
-	/// of the first message that would need to be dropped.
-	fn check_messages_footprint<'a>(
+	/// Get the `BatchFootprint` for each batch of messages `[0..n]`
+	/// as long as the total number of pages would be <= `total_pages_limit`.
```

Comment:
I think one more line about this could be useful:  
What happens when the total page limit is reached? Still the ones so far will be returned or empty return?

Reply:
Done. Added an example.